graph/resolver/schema.resolvers.go

```diff
@@ -270,8 +304,24 @@ func (r *mutationResolver) UpdateParticipant(ctx context.Context, discussionID s
 }
 
 func (r *mutationResolver) UpsertUserDevice(ctx context.Context, userID *string, platform model.Platform, deviceID string, token *string) (*model.UserDevice, error) {
-	// TODO: Check authz
-	resp, err := r.DAOManager.UpsertUserDevice(ctx, deviceID, userID, platform.String(), token)
+	currentUser := auth.GetAuthedUser(ctx)
+	if currentUser == nil {
+		return nil, fmt.Errorf("Need auth")
+	}
+
+	// Check if the user already has a device registered
+	userDevice, err := r.DAOManager.GetUserDeviceByUserIDPlatform(ctx, *userID, platform.String())
```

Comment:
should we support a user having multiple devices on the same platform? probably.

Reply:
Yeah good point!